src/h3/_cy/util.pyx

```diff
@@ -0,0 +1,162 @@
+from libc cimport stdlib
+from cython.view cimport array
+from .h3lib cimport H3int, H3str, h3IsValid, h3UnidirectionalEdgeIsValid
+
+cimport h3lib
+
+
+cpdef basestring c_version():
+    v = (
+        h3lib.H3_VERSION_MAJOR,
+        h3lib.H3_VERSION_MINOR,
+        h3lib.H3_VERSION_PATCH,
+    )
+
+    return '{}.{}.{}'.format(*v)
+
+
+cpdef H3int hex2int(H3str h) except 0:
+    return int(h, 16)
+
+
+cpdef H3str int2hex(H3int x):
+    """ Convert H3 integer to hex string representation
+
+    Need to be careful in Python 2 because `hex(x)` may return a string
+    with a trailing `L` character (denoting a "large" integer).
+    The formatting approach below avoids this.
+
+    Also need to be careful about unicode/str differences.
+    """
+    return '{:x}'.format(x)
+
+
+class H3ValueError(ValueError):
+    pass
+
+class H3CellError(H3ValueError):
+    pass
+
+class H3EdgeError(H3ValueError):
+    pass
+
+class H3ResolutionError(H3ValueError):
+    pass
+
+cdef check_cell(H3int h):
+    """ Check if valid H3 "cell" (hexagon or pentagon).
+
+    Does not check if a valid H3 edge, for example.
+    """
+    if h3IsValid(h) == 0:
+        raise H3CellError('Integer is not a valid H3 cell: {}'.format(hex(h)))
```

Comment:
Want to use `int2hex` here? Probably doesn't matter since this is an error message.

Reply:
Well, I wanted to use the same `check_cell()` for all interfaces (and not have to repeat code). To do that, this function happens at the C/Cython level, and doesn't know if it is being called by an interface that represents `H3Index`s as `str`s or `int`s.

I figured I basically had these three options:

<img width="570" alt="Screen Shot 2020-05-09 at 2 59 55 PM" src="https://user-images.githubusercontent.com/984039/81485894-c4a71f00-9205-11ea-9581-8da72a2053a9.png">

Of those, the `hex()` option seemed the most informative: inform users that this **integer representation** of their input is incorrect, but with the hex representation being more informative to them if they are using a `str` interface.

Does that seem reasonable?

Alternatively, I guess we could use `int2hex()` and have the output string prepend `0x`. But not sure if there's a benefit there.

Reply:
The string representation of the H3 indexes is hexadecimal without a prefixed `0x` though.

Reply:
Also added a note in the function docstring.

Reply:
> The string representation of the H3 indexes is hexadecimal without a prefixed `0x` though.

Yes, my thinking is that `0x...` is more "correct" in terms of the language and integer representation.

Reply:
Why would you deviate from what every other binding and the C code itself is doing?

Reply:
I think one of the points of this re-write was to deviate from the other bindings. ;)
At least, in the fact that we're trying to offer `uint64` interfaces as well as the otherwise standard string interface.

In that context, my thinking is that the fundamentally "correct" and unambiguous representation of an `H3Index` is its `uint64` representation. I realize that the `str` representation is the one people are used to, which is why I was trying to strike a balance in this error message. 

It's **correct** in the sense that the error is saying "The integer `0x...` that i received is not a valid H3Index", but it is **familiar** in the sense that a user sees the hex representation.

Suppose, for example, a user comes to this library at some point in the future and only uses the `numpy_int` interface. When they see an error that hex `'f'` is invalid, they might think "What!? I never put in 'f'!", but if they see that hex `0xf` is invalid, they have a better idea that it is talking about an int.

Reply:
I guess an alternative approach would be to have custom error message formatting for each interface. Maybe that's a good future issue to tackle?